dll/win32/shell32/folders/CDrivesFolder.cpp

```diff
@@ -410,22 +465,15 @@ HRESULT CDrivesContextMenu_CreateInstance(PCIDLIST_ABSOLUTE pidlFolder,
 }
 
 static HRESULT
-getIconLocationForDrive(IShellFolder *psf, PCITEMID_CHILD pidl, UINT uFlags,
-                        LPWSTR szIconFile, UINT cchMax, int *piIndex, UINT *pwFlags)
+getIconLocationForDrive(PCWSTR pszDrivePath, PWSTR szIconFile, UINT cchMax, int *piIndex, UINT *pGilOut)
```

Comment:
It would be of interest to start using SAL2 annotations in the shell.
```suggestion
getIconLocationForDrive(
    _In_ PCWSTR pszDrivePath,
    _Out_writes_z_(cchMax) PWSTR szIconFile,
    _In_ UINT cchMax,
    _Out_ int *piIndex,
    _Out_ UINT *pGilOut)
```

Reply:
What's this `pGilOut` ? It doesn't seem to be used here (and in the other `GetRegCustomizedDriveIcon` function).
Note that where this function is invoked down below (~ line 584), we still use a `&flags` .

Reply:
Flags is meaningless and confusing. Some parts of the shell gets these flags wrong so I have started to be very specific (mostly in my coming icon branch). [`IExtractIconA::GetIconLocation`](https://learn.microsoft.com/en-us/windows/win32/api/shlobj_core/nf-shlobj_core-iextracticona-geticonlocation) has `GIL_*` flags, some are input flags and some are output.

Reply:
I still want to name these GilIn and GilOut even with SAL.

Reply:
The GilIn/Out naming is fine; as I mentioned in the comment above I noticed that you indeed changed elsewhere "flags" into "gilout", so now the naming is consistent.